test-data/unit/check-expressions.test

```diff
@@ -2303,6 +2303,17 @@ if f == 0:  # E: Non-overlapping equality check (left operand type: "FileId", ri
     ...
 [builtins fixtures/bool.pyi]
 
+[case testStrictEqualityPromotionsLiterals]
+# flags: --strict-equality --py2
+from typing import Final
+
+U_FOO = u'foo'  # type: Final
+
+if str() == U_FOO:
+    pass
+assert u'foo' == 'foo'
```

Comment:
Maybe we can also add a line ensuring something like `assert u'foo' == u'bar'` is still flagged as an error?